sonic-chassisd/scripts/chassisd

```diff
@@ -167,6 +168,59 @@ class ModuleConfigUpdater(logger.Logger):
             self.log_info("Changing module {} to admin {} state".format(key, 'DOWN' if admin_state == MODULE_ADMIN_DOWN else 'UP'))
             try_get(self.chassis.get_module(module_index).set_admin_state, admin_state, default=False)
 
+#
+# SmartSwitch Module Config Updater ========================================================
+#
+
+
+class SmartSwitchModuleConfigUpdater(logger.Logger):
+
+    def __init__(self, log_identifier, chassis):
+        """
+        Constructor for SmartSwitchModuleConfigUpdater
+        :param chassis: Object representing a platform chassis
+        """
+        super(SmartSwitchModuleConfigUpdater, self).__init__(log_identifier)
+
+        self.chassis = chassis
+        # The threading.Lock is required for concurrent setting of admin_state
+        # on multiple DPUs for faster bring-up and shutdown.
+        self.lock = threading.Lock()
+
```

Comment:
@rameshraghupathy which data is be protected against multiple thread access?

Reply:
@prgeor It is hard to predict and say what vendor API would do under the "submit_callback". When multiple threads are forked chassis/modules resources can be corrupted. One thread can increase the chassis.number_of_modules and the other can decrease it causing data corruption and unpredictable states and behavior.  In fact it is good have another one in submit_callback as well.  But I haven't tested putting a lock in submit_callback.